src/iosource/Manager.cc

```diff
@@ -371,10 +371,6 @@ PktSrc* Manager::OpenPktSrc(const std::string& path, bool is_live)
 	PktSrc* ps = (*component->Factory())(npath, is_live);
 	assert(ps);
 
-	if ( ! ps->IsOpen() && ps->IsError() )
-		// Set an error message if it didn't open successfully.
-		ps->Error("could not open");
```

Comment:
I don't see a replacement for this error to catch & report when an open failed, is that handled elsewhere already?

Reply:
The callers of `OpenPktSrc` will report an error for a not-open source:

https://github.com/zeek/zeek/blob/2000e2a4245632ca4a5018e8ec1497652d566502/src/Net.cc#L159-L164

It's true that a source that is actually open, but happened to set an error message, will not report that error, but that's not different than the previous behavior.  The whole "Error" API for PktSrc is weird in this sense of not immediately reporting errors and only reporting an error that happens when initially trying to open it.  Not against changing it, but it's getting out of scope of the original problem and could involve changing semantics of the "Error" API when really someone can just workaround the weirdness by calling `reporter` methods directly if they want messages to immediately be displayed.

But this particular piece of "could not open" code that I removed is actively Bad (see the individual commit message 4f011a65f4313e8dcf351900b7570ff5ab7b0bc0) and I suggest removing it ASAP since it would overwrite an error message that the PktSrc set with an arbitrary one that's less useful (`ps->Error(msg)` does not _emit_ a message, it _sets_ the message).